pint/models/dispersion_model.py

```diff
@@ -27,19 +27,59 @@ def __init__(self):
         super(Dispersion, self).__init__()
         self.add_param(p.floatParameter(name="DM",
                        units="pc cm^-3", value=0.0,
-                       description="Dispersion measure"))
-        self.dm_value_funcs = [self.constant_dm,]
+                       description="Dispersion measure", long_double=True))
+        self.add_param(p.prefixParameter(name="DM1", value=0.0, units='pc cm^-3/yr^1',
+                       description="1'th time derivative of the dispersion measure",
+                       unit_template=self.DM_dervative_unit,
+                       description_template=self.DM_dervative_description,
+                       type_match='float', long_double=True))
+        self.add_param(p.MJDParameter(name="DMEPOCH",
+                       description="Epoch of DM measurement"))
+
+        self.dm_value_funcs = [self.base_dm,]
         self.delay_funcs['L1'] += [self.dispersion_delay,]
         self.order_number = 2
+        self.print_par_func = 'print_par_DMs'
 
     def setup(self):
         super(Dispersion, self).setup()
-        self.register_deriv_funcs(self.d_delay_d_DM, 'delay', 'DM')
+        # If DM1 is set, we need DMEPOCH
+        if self.DM1.value != 0.0:
+            if self.DMEPOCH.value is None:
+                raise MissingParameter("Dispersion", "DMEPOCH",
+                        "DMEPOCH is required if DM1 or higher are set")
+        base_dms = self.get_prefix_mapping('DM').values()
```

Comment:
In Python3, dict.values() apparently doesn't return a list.  Try adding list() around this statement.